### PR TITLE
(FM-7869) Implement a Remote Task

### PR DESCRIPTION
**Pull request by da-ar**

### CODE REVIEW

File: lib/ace/transport_app.rb [hunk 1]
```diff
@@ -49,17 +61,25 @@ def validate_schema(schema, body)
...
 
-      # grab the transport connection_info
-      connection_info = Hash[body['target'].map { |k, v| [k.to_sym, v] }] # may contain sensitive data
+      opts = body['target'].merge('protocol' => 'remote')
 
-      # originally this was a Bolt::Task::PuppetServer; simplified here for hacking
-      task = Bolt::Task.new(body['task'])
+      # This is a workaround for Bolt due to the way it expects to receive the target info
+      # see: https://github.com/puppetlabs/bolt/pull/915#discussion_r268280535
+      # Systems calling into ACE will need to determine the address/uri/ip and pass this as `name`
```
Comment (DavidS) [inline lib/ace/transport_app.rb]: `... will need to determine the nodename/certname and pass this...`

File: lib/ace/transport_app.rb [hunk 2]
```diff
@@ -2,8 +2,12 @@
 
 require 'ace'
 require 'ace/executor'
+require 'ace/task/remote_server'
```
Comment (Thomas-Franklin) [inline lib/ace/transport_app.rb]: This appears to be non existing

**Comment by willmeek** [comment]
Travis failing due to that non existing require, but looks good

File: lib/ace/transport_app.rb [hunk 3]
```diff
@@ -2,8 +2,11 @@
 
 require 'ace'
 require 'ace/executor'
```
Comment (Thomas-Franklin) [inline lib/ace/transport_app.rb]: The ace/executor is no longer required

File: spec/ace/config_spec.rb [hunk 4]
```diff
@@ -0,0 +1,176 @@
+# frozen_string_literal: true
```
Comment (Thomas-Franklin) [inline spec/ace/config_spec.rb]: These tests are part of #17 so I do not think they are required as part of this PR as it is not related to remote tasks, or close #17 and move this to spec/unit/ace folder

Comment (da-ar) [reply spec/ace/config_spec.rb]: This work is part of the completion of remote tasks. I did miss that https://github.com/puppetlabs/ace/pull/17 was up, but as it is a direct port of the Bolt test it doesn't give us much. The changes made here provide us with the additional testing required. I would suggest closing https://github.com/puppetlabs/ace/pull/17

This continues to be WIP, I've updated the title as such.

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -16,7 +16,8 @@
   SimpleCov.start do
     track_files 'lib/**/*.rb'
 
-    add_filter 'lib/puppet/resource_api/version.rb'
+    add_filter 'lib/ace/version.rb'
+    add_filter 'lib/ace.rb'
```
Comment (DavidS) [inline spec/spec_helper.rb]: this file does not exist, and `lib/ace.rb` above can be deleted if it's empty. 

I'd rather not have those landmines lying about here.

**Pull request merged**